servers/quarkus-server/src/main/java/org/projectnessie/server/providers/ResteasyExceptionMapper.java

```diff
@@ -15,44 +15,59 @@
  */
 package org.projectnessie.server.providers;
 
-import javax.enterprise.inject.Alternative;
+import javax.annotation.Priority;
+import javax.inject.Inject;
 import javax.validation.ValidationException;
+import javax.ws.rs.Priorities;
 import javax.ws.rs.core.Response;
 import javax.ws.rs.core.Response.Status;
 import javax.ws.rs.ext.ExceptionMapper;
 import javax.ws.rs.ext.Provider;
 
 import org.jboss.resteasy.api.validation.ResteasyViolationException;
 import org.jboss.resteasy.api.validation.Validation;
+import org.projectnessie.services.config.ServerConfig;
+import org.projectnessie.services.rest.BaseExceptionMapper;
+import org.projectnessie.services.rest.NessieExceptionMapper;
 
 /**
  * "Special" implementation for exceptions that extend {@link ValidationException}, as those
  * do not "go through" {@link NessieExceptionMapper} and there need to be two
  * {@link ExceptionMapper} beans for the Nessie-server.
  */
 // Use our exception-mapper instead of io.quarkus.hibernate.validator.runtime.jaxrs.ResteasyViolationExceptionMapper
-@Alternative
+@Priority(Priorities.USER)
 @Provider
 public class ResteasyExceptionMapper
     extends BaseExceptionMapper
-    implements ExceptionMapper<ValidationException> {
+    implements ExceptionMapper<ResteasyViolationException> {
+
+
+  // Unused constructor
+  // Required because of https://issues.jboss.org/browse/RESTEASY-1538
+  public ResteasyExceptionMapper() {
+    this(null);
+  }
+
+  @Inject
+  public ResteasyExceptionMapper(ServerConfig config) {
+    super(config);
+  }
 
   @Override
-  public Response toResponse(ValidationException exception) {
-    if (exception instanceof ResteasyViolationException) {
-      ResteasyViolationException violationException = (ResteasyViolationException) exception;
-      Exception e = violationException.getException();
-      if (e == null) {
-        boolean returnValueViolation = !violationException.getReturnValueViolations().isEmpty();
-        Status st = returnValueViolation ? Status.INTERNAL_SERVER_ERROR : Status.BAD_REQUEST;
-        return buildExceptionResponse(
-            st.getStatusCode(),
-            st.getReasonPhrase(),
-            exception.getMessage(),
-            exception,
-            false, // no need to send the stack trace for a validation-error
-            b -> b.header(Validation.VALIDATION_HEADER, "true"));
-      }
+  public Response toResponse(ResteasyViolationException exception) {
+    ResteasyViolationException violationException = exception;
```

Comment:
why do this?

Reply:
Because exception violations in quarkus uses a specific subclass of `ValidationException` and so to keep the logic which was present before?

Reply:
this specifically is wierd:

> ResteasyViolationException violationException = exception; 

isnt this just renaming a variable? Is the goal to just keep the diff size down or is thre reassignment something subtle I am missing

Reply:
aah, thanks for catching, yes, a bit more cleanup is needed :)